src/GovUk.Education.ExploreEducationStatistics.Content.Services.Tests/PublicationServiceTests.cs

```diff
@@ -16,2039 +16,2273 @@
+                GenerateTheme(publishedPublications.Skip(1), supersededPublications.Skip(1), publishedPublicationsWithUnpublishedSuperseded.Skip(2))
+            };
 
-                var publicationB = new Publication
-                {
-                    Title = "Publication B",
-                    LatestPublishedReleaseVersion = new ReleaseVersion
-                    {
-                        Type = OfficialStatistics,
-                        Published = DateTime.UtcNow
-                    }
-                };
+            await AddToDatabase(themes);
+            
+            // Act
+            var results = await ListPublicationInfos();
+
+            // Assert
+            var expectedPublications = publishedPublications
+                .Concat(publishedPublicationsWithUnpublishedSuperseded)
+                .Concat(supersededPublications.Select(p => p.SupersededBy).OfType<Publication>())
+                .ToArray();
+            
+            AssertPublicationInfosAsExpected(expectedPublications, results);
+        }
 
-                var publicationC = new Publication
-                {
-                    Title = "Publication C",
-                    LatestPublishedReleaseVersion = new ReleaseVersion
-                    {
-                        Type = AdHocStatistics,
-                        Published = DateTime.UtcNow
-                    }
-                };
+        [Fact]
+        public async Task GivenPublications_WhenListPublicationInfosByThemeIs_ThenReturnsPublicationInfosInTheme()
+        {
+            // Arrange
+            var publishedPublications = 
+                Enumerable
+                    .Range(1, 6)
+                    .Select(_ => GeneratePublishedPublication())
+                    .ToArray();
+            
+            var unpublishedPublications = 
+                Enumerable
+                    .Range(1, 6)
+                    .Select(_ => GenerateUnpublishedPublication())
+                    .ToArray();
+            
+            var publishedPublicationsWithUnpublishedSuperseded = 
+                Enumerable
+                    .Range(1, 6)
+                    .Select(_ => GeneratePublicationWithUnpublishedSuperceded())
+                    .ToArray();
+            
+            var supersededPublications = 
+                Enumerable
+                    .Range(1, 6)
+                    .Select(_ => GeneratePublicationWithPublishedSuperceded())
+                    .ToArray();
+
+            var allPublications = publishedPublications
+                .Concat(unpublishedPublications)
+                .Concat(publishedPublicationsWithUnpublishedSuperseded)
+                .Concat(supersededPublications)
+                .Concat(supersededPublications.Select(p => p.SupersededBy).OfType<Publication>())
+                .ToArray()
+                .Shuffle(seed:12345);
+
+            var theme1Publications = allPublications.Take(allPublications.Length / 3).ToArray();
```

Comment:
It's not immediately clear why the need to be dividing by 3, multiplying by 2 in this test.  It might also make it difficult for people to add new Publications to the test setup in the future if it then throws off the maths.  Might be simpler to go a bit more boilerplate-y here?  Happy to get a second opinion!